From 4b26b877762f35a134a9afbff5f8c30c0856326c Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Tue, 21 Jul 2026 20:11:49 +0000
Subject: [PATCH 138/183] dt-bindings: net: use xPON dying-gasp interrupt

Document the optional dedicated power-fail interrupt exposed by the xPON
block and remove the redundant omci-dying-gasp boolean. The interrupt
itself now describes whether the platform can deliver dying-gasp events.

Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 .../devicetree/bindings/net/airoha,en7523-xpon.yaml         | 3 +++
 Documentation/devicetree/bindings/net/omci.yaml             | 6 ------
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml b/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
index fa6366fad770..94560a47c06a 100644
--- a/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
+++ b/Documentation/devicetree/bindings/net/airoha,en7523-xpon.yaml
@@ -43,6 +43,9 @@ properties:
   interrupts:
     minItems: 1
     maxItems: 2
+    description:
+      The first interrupt is generated by the xPON MAC. The optional second
+      interrupt reports the platform power-fail/dying-gasp condition.
 
   interrupt-names:
     minItems: 1
diff --git a/Documentation/devicetree/bindings/net/omci.yaml b/Documentation/devicetree/bindings/net/omci.yaml
index 49e372f669b2..569b0ed12645 100644
--- a/Documentation/devicetree/bindings/net/omci.yaml
+++ b/Documentation/devicetree/bindings/net/omci.yaml
@@ -84,11 +84,5 @@ properties:
         minItems: 1
         maxItems: 20
 
-  omci-dying-gasp:
-    type: boolean
-    description:
-      Enable a best-effort ONU-G dying-gasp alarm notification over OMCI in
-      addition to the GPON PLOAM dying-gasp message. Disabled by default.
-
 additionalProperties: true
 ...
-- 
2.55.0

